torch/csrc/api/include/torch/nn/modules/transformercoder.h

```diff
@@ -0,0 +1,71 @@
+#pragma once
+
+#include <torch/nn/cloneable.h>
+#include <torch/nn/module.h>
+#include <torch/nn/modules/container/any.h>
+#include <torch/nn/modules/container/modulelist.h>
+#include <torch/nn/options/transformercoder.h>
+#include <torch/nn/pimpl.h>
+#include <torch/nn/modules/common.h>
+
+#include <torch/types.h>
+
+#include <ostream>
+
+namespace torch {
+namespace nn {
+
+// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ TransformerEncoder ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+/// TransformerEncoder module.
+/// See https://pytorch.org/docs/master/generated/torch.nn.TransformerEncoder.html to
+/// learn abouut the exact behavior of this encoder layer module.
+///
+/// See the documentation for `torch::nn::TransformerEncoder` class to learn what
+/// constructor arguments are supported for this encoder module.
+///
+/// Example:
+/// ```
+/// TransformerEncoderLayer encoderLayer(TransformerEncoderLayerOptions(512, 8).dropout(0.1));
+//  TransformerEncoder encoder(TransformerEncoderOptions(encoderLayer, 6).norm(LayerNorm(LayerNormOptions({2}))));
+/// ```
+class TORCH_API TransformerEncoderImpl : public Cloneable<TransformerEncoderImpl> {
+
+  public:
+    explicit TransformerEncoderImpl(TransformerEncoderOptions options_);
+
+    Tensor forward(
+      const Tensor& src,
+      const Tensor& src_mask = {},
```

Comment:
The second argument is called "mask" in Python: https://github.com/pytorch/pytorch/blob/cbdaa20c88306b9a828edd6e766d3c987b19a628/torch/nn/modules/transformer.py#L167. Does it matter that this one is different?

I do prefer `src_mask` because it makes it clear what is getting masked...

Reply:
@zhangguanheng66 I make the name to src_mask for the same reason as @zou3519, agree?

Reply:
I think it makes sense to call `src_mask` and `tgt_mask` along with `src_key_padding_mask` and `tgt_key_padding_mask` (consistent with the python ones)